search-provider/eks-metadata-provider-dbus.xml

```diff
@@ -0,0 +1,113 @@
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+
+<!--
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General
+ Public License along with this library; if not, see <http://www.gnu.org/licenses/>.
+-->
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <interface name="com.endlessm.Metadata">
+    <!--
+        Query:
+        @Query: A dictionary describing the query to be made, with the following
+                optional parameters. Specifying any parameter that is not
+                a member of this list is an error. No new parameters will
+                be added or removed from this interface during its lifetime.
+
+                "search-terms": A string (s) with the free-text search terms
+                                to use during the query. If the parameter
+                                is not specified, filtering will only
+                                be done against tags-match-any and tags-match-all.
+                "tags-match-any": A strv (as) with a list of tags that any
+                                  matching content objects must have at least
+                                  one of.
+                "tags-match-all": A strv (as) with a list of tags that any
+                                  matching content objects must have all of.
+                "limit": An integer indicating the number of results at most
+                         to return. If the parameter is not specifed there
+                         will be no limit.
+                "offset": An integer specifying the "offset" into the query
+                          results from the point which it start returning results.
+                          This also changes the upper bound if "limit" was
```

Comment:
To follow on from my previous comment, I got confused because "upper bound" already has a different meaning in `Xapian::MSet` and `DmQueryResult`. It would be good to use a different term here.